profiler/src/ProfilerEngine/Datadog.Profiler.Native/RawSamples.hpp

```diff
@@ -0,0 +1,69 @@
+
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+
+#pragma once
+#include <list>
+
+template <class TRawSample>
+class RawSamples
+{
+public:
+    using const_iterator = typename std::list<TRawSample>::const_iterator;
+
+    RawSamples() = default;
+
+    RawSamples(RawSamples&& other)
+    {
+        *this = std::move(other);
+    };
+
+    RawSamples& operator=(RawSamples&& other)
+    {
+        _samples = std::move(other._samples);
+        return *this;
+    }
+
+    RawSamples(RawSamples const&) = delete;
+    RawSamples& operator=(RawSamples const& other) = delete;
+
+    RawSamples<TRawSample> FetchRawSamples()
```

Comment:
rename to "Move" for consistency (I liked your Steal() idea  :^)

Reply:
✔️